src/AutoRest.CSharp/Common/Input/InputTypes/InputPrimitiveType.cs

```diff
@@ -3,37 +3,37 @@
 
 namespace AutoRest.CSharp.Common.Input;
 
-internal record InputPrimitiveType(InputTypeKind Kind, bool IsNullable = false) : InputType(Kind.ToString(), IsNullable)
+internal record InputPrimitiveType(InputTypeKind Kind, bool IsConfident, bool IsNullable = false) : InputType(Kind.ToString(), IsConfident, IsNullable: IsNullable)
```

Comment:
Can we have the default value of 'IsConfident' since most of the cases, this value is true.

Reply:
In general, as we are confident about the most of the features, can we have the default value of true in all inputTypes?

Reply:
No. Making a property optional has more possibilities to get it wrong values when we construct them.
We have done that for the `IsNullable` property at many places.

These constructors will nearly never be used other than deserialization, therefore making all of them "REQUIRED" has more benefits than making it optional as you said to have a default value which ensures we ALWAYS explicitly set them

Reply:
BTW I changed IsNullable to required as well in this PR: https://github.com/Azure/autorest.csharp/pull/3373
The reason of that is that these values are important therefore we should always ensure we are setting them explicitly even if in most cases they are having the same value to avoid some hidden issues which I discovered a few in the nullable PR.
For instance [this issue](https://github.com/Azure/autorest.csharp/issues/3600) is exactly caused by the default value we set, and we forgot to deserialize its `IsNullable` value when deserializing the tspCodeModel and the compiler does not give us any warning about that.